src/Routes/Devices/UpdateDeviceModal.js

```diff
@@ -53,29 +59,63 @@ const UpdateDeviceModal = ({ updateModal, setUpdateModal, refreshTable }) => {
           setImageData(data.Data.images[0])
         );
   }, []);
+  const statusMessages = {
+    onInfo: {
+      title: 'Updating system',
+      description: isMultiple
+        ? ` ${deviceName.length} systems were added to the queue.`
+        : ` ${deviceName} was added to the queue.`,
+    },
+    onError: {
+      title: 'Error',
+      description: `Failed to update the selected system(s)`,
+    },
+  };
 
   const handleUpdateModal = async () => {
-    try {
-      await updateSystem({
-        DevicesUUID: deviceId,
-      });
-      dispatch({
-        ...addNotification({
-          variant: 'info',
-          title: 'Updating system',
-          description: isMultiple
-            ? ` ${deviceName.length} systems were added to the queue.`
-            : ` ${deviceName} was added to the queue.`,
-        }),
-      });
-    } catch (err) {
-      dispatch({
-        ...addNotification({
-          variant: 'danger',
-          title: 'Updating a system was unsuccessful',
-          description: `Response: ${err.statusText}`,
-        }),
-      });
+    if (notificationProp) {
+      try {
+        await apiWithToast(
+          dispatch,
+          () =>
+            updateSystem({
+              DevicesUUID: deviceId,
+            }),
+          statusMessages,
+          notificationProp
+        );
+      } catch (err) {
+        dispatch({
+          ...addNotification({
+            variant: 'danger',
+            title: 'Updating a system was unsuccessful',
+            description: `Response: ${err.statusText}`,
+          }),
+        });
+      }
+    } else {
```

Comment:
@mgold1234 why is it necessary to check if the `notificationProp` exists?

Reply:
I'm asking because `apiWithToast` already checks it and switches to use or not `notificationProp`

Reply:
its not working without it, we need to use the notification that we catch in case we work on federation mode

Reply:
I understand the `notificationProp` is important for the federation mode
What I'm trying to say is, that you might not need the `if..else` condition because if the `notificationProp` is `undefined` the `apiWithToast` is already handling this case

Reply:
check here
https://github.com/RedHatInsights/edge-frontend/blob/master/src/utils/apiWithToast.js#L25-L33